segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/stat/AutoScaleProcessor.java

```diff
@@ -111,7 +113,18 @@ private void bootstrapRequestWriters() {
                 })
                 .runAsync(() -> {
                     if (clientFactory.get() == null) {
-                        clientFactory.compareAndSet(null, ClientFactory.withScope(NameUtils.INTERNAL_SCOPE_NAME, configuration.getControllerUri()));
+                        ClientFactory factory = null;
+                        if (configuration.isAuthEnabled()) {
+                            factory = ClientFactory.withScope(NameUtils.INTERNAL_SCOPE_NAME,
+                                    PravegaClientConfig.builder().controllerURI(configuration.getControllerUri())
+                                    .credentials(new PravegaDefaultCredentials(configuration.getAuthPassword(), configuration.getAuthUsername()))
+                                    .pravegaTrustStore(configuration.getTlsCertFile())
```

Comment:
Could there be a field in the configuration object that maps to the validateHostName field of PravegaClientConfig?  Such that one can explicitly disable it?   And this would call `.validateHostName(configuration.getValidateHostName()`?

In deployments where certs are not using real host names, both external clients, and this internal communication, will have trouble with host validation.

Reply:
Yes, this makes sense. The default value will be host name enabled. It can be overridden with a config. This part is going to change once we implement the changes for not depending on default password. Created a new issue #2336 to track this.